modules/apps/callbacks/v2/ibc_middleware.go

```diff
@@ -14,9 +14,29 @@ import (
 	channeltypes "github.com/cosmos/ibc-go/v10/modules/core/04-channel/types"
 	channeltypesv2 "github.com/cosmos/ibc-go/v10/modules/core/04-channel/v2/types"
 	"github.com/cosmos/ibc-go/v10/modules/core/api"
+	"github.com/cosmos/ibc-go/v10/modules/core/exported"
 )
 
-var _ api.IBCModule = (*IBCMiddleware)(nil)
+var (
+	_ api.IBCModule            = (*IBCMiddleware)(nil)
+	_ exported.Acknowledgement = (*RecvAcknowledgement)(nil)
+)
+
+// Create internal implementation of exported.Acknowledgement
+// to pass the app acknowledgement bytes to contractKeeper IBCReceivePacketCallback
+// interface
+type RecvAcknowledgement []byte
+
+// RecvPacket only passes callback to contract if the acknowledgement
+// is successful. Thus, we can just return true here.
+func (rack RecvAcknowledgement) Success() bool {
+	return true
```

Comment:
This is not we did with `channeltypes2.Acknowledgement`? Don't we do this check: https://github.com/cosmos/ibc-go/blob/75c3eed2d57e287fd46d64a6e436cec06c69768b/modules/core/04-channel/v2/types/acknowledgement.go#L42

Reply:
Yes, im just satisfying the interface since it is guaranteed to not be ErrorAcknowledgement here

Reply:
Yes, but why don't we add this check anyway. If we change the callback behavior in the future (like making callbacks on failures as well), adding this would save us.

Reply:
Ok i can add